server/middlewares/mainValidation.js

```diff
@@ -0,0 +1,154 @@
+import Joi from 'joi';
+import partyObj from '../dummyData/partyObj';
+import officeObj from '../dummyData/officeObj';
+import {
+  newPartySchema, idSchema, newOfficeSchema, officeIdSchema
+} from '../inputSchema';
+
+class mainValidator {
+  static newPartyHelper(request, response, next) {
+    const { email, name } = request.body;
+    const { error } = Joi.validate(request.body, newPartySchema, { abortEarly: false });
+    if (error !== null) {
+      response.status(400)
+        .json({
+          success: false,
+          message: error.details.map(d => d.message)
+        });
+      return false;
+    }
+    const dupEmail = partyObj.find(party => party.email === email);
+    const dupName = partyObj.find(party => party.name === name);
+
+    if (dupEmail !== undefined || dupName !== undefined) {
+      response.status(409)
+        .json({
+          success: false,
+          message: 'Email or name already exist'
+        });
+      return false;
+    }
+    next();
+  }
+
+  static getPartyHelper(request, response, next) {
+    const { partyId } = request.params;
+    const { error } = Joi.validate(request.params, idSchema, { abortEarly: false });
+    if (error !== null) {
+      response.status(400)
+        .json({
+          success: false,
+          message: error.details.map(d => d.message)
+        });
+      return false;
+    }
+    const partyExist = partyObj.find(party => party.id === Number(partyId));
```

Comment:
Not found errors should be handled from the controller. Validation errors are errors due to invalid data from the user. This is not one of such. Also, why is this using `find()` and the implementation for deleting using `findIndex()`?

On a general note, let your validators be concerned only about ensuring that user inputs are valid.